paymentsheet/src/main/java/com/stripe/android/link/LinkActivityViewModel.kt

```diff
@@ -117,12 +118,13 @@ internal class LinkActivityViewModel @Inject constructor(
         navController ?: return
         navListenerJob = viewModelScope.launch {
             navController.currentBackStackEntryFlow.collectLatest { entry ->
+                val previousEntry = navController.previousBackStackEntry
                 val route = entry.destination.route
                 _linkAppBarState.update {
                     it.copy(
                         showHeader = showHeaderRoutes.contains(route),
                         showOverflowMenu = route == LinkScreen.Wallet.route,
-                        navigationIcon = if (route == LinkScreen.PaymentMethod.route) {
+                        navigationIcon = if (previousEntry != null) {
```

Comment:
Since we use `navigateAndClearStack` whenever appropriate, we can rely on the `previousBackStackEntry`.